### PR TITLE
Associated Wallets

### PR DESCRIPTION
**Pull request by jowlee**
### Description
Adds support for associated wallets. 
* Updates the /audio page UI to display the user's wallet total and total across all wallets.
* Updates the /audio page UI to add a modal for managing the user's associated wallets
* Removes claim logic from the UI
* Adds several dependencies for web3 providers to allow users to connect their wallet
  * `web3modal` The parent package to manage other web3 providers given a config
  * `bitski`, `@toruslabs/torus-embed`, `@walletconnect/web3-provider`, `eth-provider` are all web3 provider options
* Update badge to be based on total balance across all wallets - not just owner wallet
* Updated AudiusAPIClient to handle both /full and non full routes 
* Added `getAssociatedWallets` and `getAssociatedWalletUserId` to the AudiusAPIClient
 

### Dragons
none

### How Has This Been Tested?
I ran everything locally against staging.

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://frontend-demo.audius.co/jowlee-associated-wallets

File: src/containers/audio-rewards-page/WalletModal.module.css [hunk 1]
```diff
@@ -4,6 +4,14 @@
   width: 100%;
 }
 
+.modalBody.wallets {
```
Comment (sddioulde) [inline src/containers/audio-rewards-page/WalletModal.module.css:7]: why '.modalBody.wallets' twice?

File: src/store/token-dashboard/slice.ts [hunk 2]
```diff
@@ -119,8 +144,94 @@ const slice = createSlice({
...
+        state.associatedWallets.connectedWallets || []
+      ).concat(wallet)
+    },
+    setWalletRemovedConfirmed: (
+      state,
+      { payload: { wallet } }: PayloadAction<{ wallet: string }>
+    ) => {
+      state.associatedWallets.connectedWallets = (
+        state.associatedWallets.connectedWallets || []
+      ).filter(connectedWallet => connectedWallet === wallet)
```
Comment (sddioulde) [inline src/store/token-dashboard/slice.ts:191]: should the filter be NOT_EQUAL i.e. connectedWallet => connectedWallet !== wallet?

Comment (jowlee) [reply src/store/token-dashboard/slice.ts:191]: thx, yeah I don't even use this - will delete

File: src/containers/audio-rewards-page/WalletModal.module.css (hunk 1 above)
Comment (jowlee) [reply src/containers/audio-rewards-page/WalletModal.module.css:7]: Screen Shot 2021-03-25 at 3 53 00 PM
I wanted this selector to be more specific if the modal body was displaying wallets to modify the width

Comment (jowlee) [reply src/containers/audio-rewards-page/WalletModal.module.css:7]: Oh, I misread the comment, yeah will merge into one

File: src/store/token-dashboard/slice.ts [hunk 3]
```diff
@@ -182,14 +182,6 @@ const slice = createSlice({
         state.associatedWallets.connectedWallets || []
       ).concat(wallet)
     },
-    setWalletRemovedConfirmed: (
```
Comment (sddioulde) [inline src/store/token-dashboard/slice.ts]: nice

File: src/containers/audio-rewards-page/WalletModal.module.css [hunk 4]
```diff
@@ -6,9 +6,6 @@
 
 .modalBody.wallets {
   max-width: 720px;
-}
```
Comment (sddioulde) [inline src/containers/audio-rewards-page/WalletModal.module.css]: cool

**Review by sddioulde — APPROVED** [review]

**Pull request merged**